js/modules/Helpdesk/HelpdeskConfigController.js

```diff
@@ -0,0 +1,263 @@
+            })
+        ;
+
+        this.#container
+            .querySelectorAll('[data-glpi-helpdesk-config-action-delete')
+            .forEach((node) => {
+                node.addEventListener('click', (e) => {
+                    const tile = e.target.closest('[data-glpi-helpdesk-config-tile-container]');
+                    this.#deleteTile(tile);
+                });
+            })
+        ;
+    }
+
+    #showReorderUI()
+    {
+        this.#container
+            .querySelector('[data-glpi-helpdesk-config-reorder-actions]')
+            .classList
+            .remove('d-none')
+        ;
+        this.#container
+            .querySelectorAll('[data-glpi-helpdesk-config-extra-actions]')
+            .forEach((dots) => {
+                dots.classList.add('d-none');
+            })
+        ;
+        this.#container
+            .querySelectorAll('[data-glpi-helpdesk-config-tile]')
+            .forEach((tile_body) => {
+                tile_body.classList.add('border-2');
+                tile_body.classList.add('border-dashed');
+            })
+        ;
+    }
+
+    #hideReorderUI()
+    {
+        this.#container
+            .querySelector('[data-glpi-helpdesk-config-reorder-actions]')
+            .classList
+            .add('d-none')
+        ;
+        this.#container
+            .querySelectorAll('[data-glpi-helpdesk-config-extra-actions]')
+            .forEach((dots) => {
+                dots.classList.remove('d-none');
+            })
+        ;
+        this.#container
+            .querySelectorAll('[data-glpi-helpdesk-config-tile]')
+            .forEach((tile_body) => {
+                tile_body.classList.remove('border-2');
+                tile_body.classList.remove('border-dashed');
+            })
+        ;
+    }
+
+    async #reloadTiles()
+    {
+        try {
+            const url = `${CFG_GLPI.root_doc}/Config/Helpdesk/FetchTiles`;
+            const url_params = new URLSearchParams({
+                profile_id: this.#profile_id,
+            });
+            const response = await fetch(`${url}?${url_params}`);
+            if (!response.ok) {
+                throw new Error(response.status);
+            }
+
+            this.#getTilesContainerDiv().innerHTML = await response.text();
+        } catch (e) {
+            glpi_toast_error(__('An unexpected error occurred.'));
+            console.error(e);
+        }
+    }
+
+    async #saveTilesOrder()
+    {
+        try {
+            // Set up form data
+            const form_data = new FormData();
+            form_data.append('profile_id', this.#profile_id);
+            this.#getTilesOrder()
+                .forEach((id) => form_data.append("order[]", id))
+            ;
+
+            // Send request
+            const url = `${CFG_GLPI.root_doc}/ajax/Config/Helpdesk/SetTilesOrder`;
+            const response = await fetch(url, {
+                method: 'POST',
+                body: form_data,
+                headers: {
```

Comment:
Will be necessary for #18723.

```suggestion
                headers: {
                    'X-Requested-With': 'XMLHttpRequest',
```

Reply:
It's going to be tedious to maintain this parameters everywhere.
Maybe we should just keep using jQuery for ajax queries so this + the crsf token are handled automatically ?

Reply:
We should trend to use vanilla JS when it is possible, but I agree that having defaults behaviours is easier to maintain. In a new project, I would probably use a component dedicated to requests handling, like `axios`, to be able to configure default behaviours, but as we are not going to replace jQuery for the moment, you can use it instead of the fetch API.